src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -0,0 +1,40 @@
+class Webui::Users::NotificationsController < Webui::WebuiController
+  before_action :require_login
+
+  def index
+    notification_type = params[:type]
+    case notification_type
+    when 'done'
+      @notifications = Notification.where(delivered: true)
+                                   .where("(subscriber_type = 'User' AND subscriber_id = ?) OR (subscriber_type = 'Group' AND subscriber_id IN (?))",
+                                          User.session, User.session.groups.map(&:id))
+    when 'reviews'
+      @notifications = []
+    when 'comments'
+      @notifications = Notification.where(notifiable_type: 'Comment', delivered: false)
+                                   .where("(subscriber_type = 'User' AND subscriber_id = ?) OR (subscriber_type = 'Group' AND subscriber_id IN (?))",
+                                          User.session, User.session.groups.map(&:id))
+    when 'state_changes'
+      @notifications = Notification.where(notifiable_type: 'BsRequest', delivered: false).where.not(bs_request_oldstate: nil)
+                                   .where("(subscriber_type = 'User' AND subscriber_id = ?) OR (subscriber_type = 'Group' AND subscriber_id IN (?))",
+                                          User.session, User.session.groups.map(&:id))
+    else
+      @notifications = Notification.where(delivered: false)
+                                   .where("(subscriber_type = 'User' AND subscriber_id = ?) OR (subscriber_type = 'Group' AND subscriber_id IN (?))",
+                                          User.session, User.session.groups.map(&:id))
+    end
+  end
+
+  def set_done
```

Comment:
Hm I'm not sure about this workflow. Bulk edit is something optional, why don't we start with a single action on single notifications?